Modules/Office365DSC/DSCResources/MSFT_TeamsUser/MSFT_TeamsUser.psm1

```diff
@@ -24,64 +28,77 @@ function Get-TargetResource
     )
 
     Write-Verbose -Message "Getting configuration of member $User to Team $TeamName"
 
-    Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
-        -Platform MicrosoftTeams
-
-    $nullReturn = @{
-        User               = $User
-        Role               = $Role
-        TeamName           = $TeamName
-        Ensure             = "Absent"
-        GlobalAdminAccount = $GlobalAdminAccount
-    }
-
-    Write-Verbose -Message "Checking for existance of Team User $User"
-    $team = Get-TeamByName $TeamName -ErrorAction SilentlyContinue
-    if ($null -eq $team)
+    if ($RawInputObject)
     {
-        return $nullReturn
+        $myUser = $RawInputObject
     }
+    else
+    {
+        Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
+            -Platform MicrosoftTeams
+
+        $nullReturn = @{
+            User               = $User
+            Role               = $Role
+            TeamName           = $TeamName
+            TeamMailNickName   = $TeamMailNickName
+            Ensure             = "Absent"
+            GlobalAdminAccount = $GlobalAdminAccount
+        }
 
-    Write-Verbose -Message "Retrieve team GroupId: $($team.GroupId)"
+        Write-Verbose -Message "Checking for existance of Team User $User"
+        $team = Get-TeamByName $TeamName -ErrorAction SilentlyContinue
+        if ($null -eq $team)
+        {
+            return $nullReturn
+        }
 
-    #region Telemetry
-    $data = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
-    $data.Add("Resource", $MyInvocation.MyCommand.ModuleName)
-    $data.Add("Method", $MyInvocation.MyCommand)
-    #Add-O365DSCTelemetryEvent -Data $data
-    #endregion
+        Write-Verbose -Message "Retrieve team GroupId: $($team.GroupId)"
 
-    try
-    {
-        Write-Verbose "Retrieving user without a specific Role specified"
-        $allMembers = Get-TeamUser -GroupId $team.GroupId -ErrorAction SilentlyContinue
-    }
-    catch
-    {
-        Write-Warning "The current user doesn't have the rights to access the list of members for Team {$($TeamName)}."
-        Write-Verbose $_
-        return $nullReturn
-    }
+        #region Telemetry
+        $data = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
+        $data.Add("Resource", $MyInvocation.MyCommand.ModuleName)
+        $data.Add("Method", $MyInvocation.MyCommand)
+        #Add-O365DSCTelemetryEvent -Data $data
+        #endregion
 
-    if ($null -eq $allMembers)
-    {
-        Write-Verbose -Message "Failed to get Team's users for Team $TeamName"
-        return $nullReturn
+        try
+        {
+            Write-Verbose "Retrieving user without a specific Role specified"
+            $allMembers = Get-TeamUser -GroupId $team.GroupId -ErrorAction SilentlyContinue
+        }
+        catch
+        {
+            Write-Warning "The current user doesn't have the rights to access the list of members for Team {$($TeamName)}."
```

Comment:
Did this become Verbose instead of Warning on purpose? Also on line 61. I ask this because it seems some of permission issues are displayed to users with warning message.